Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyFragment.kt

```diff
@@ -95,7 +98,11 @@ class SurveyFragment : Fragment() {
         activity?.let {
             picker.show(it.supportFragmentManager, picker.toString())
             picker.addOnPositiveButtonClickListener {
-                viewModel.onDatePicked(questionId, picker.headerText)
+                val selectedData = Date().apply {
+                    time = picker.selection ?: date
+                }
+                val dateFormatted = SimpleDateFormat("EEE, MMM d", Locale.getDefault()).format(selectedData)
+                viewModel.onDatePicked(questionId, dateFormatted)
```

Comment:
With this implementation, you're putting a lot of logic in the UI that can't be unit tested. Move this to the ViewModel.

Reply:
moved the logic into view model